src/swell/deployment/prepare_config_and_suite/prepare_config_and_suite.py

```diff
@@ -0,0 +1,601 @@
+        # in the experiment dictionary and retrieve the response
+        if 'model_components' not in self.experiment_dict.keys():
+            self.logger.abort('The model components question has not been answered.')
+
+        for model in self.experiment_dict['model_components']:
+
+            model_dict = self.question_dictionary_model_dep[model]
+
+            # Loop over keys of each model
+            for question_key in model_dict:
+
+                # Ask only the suite questions first
+                if 'suites' in model_dict[question_key].keys():
+
+                    # Ask the question
+                    self.ask_a_question(model_dict, question_key, model)
+
+        # 7. Perform a more exhaustive resolving of suite file templates
+        # --------------------------------------------------------------
+        # Note that we reset the suite file to avoid templates having been left unresolved
+        # (removed) from the previous attempt. We still do not ask for an exhaustive resolving
+        # of templates because there are things related to scheduling that are not yet able to be
+        # resolved. In the future it might be good to bring some of that information into the
+        # sphere of suite questions but that requires some careful thought so as not to overload
+        # the user with questions.
+        suite_str = template_string_jinja2(self.logger, self.suite_str, self.experiment_dict,
+                                           True)
+
+        # 8. Build a list of tasks for each model component
+        # -------------------------------------------------
+        model_dep_tasks, all_tasks = self.get_suite_task_list_model_dep(suite_str)
+
+        # 9.1 Ask the new task questions that do not actually depend on the model
+        # -----------------------------------------------------------------------
+        for question_key in self.question_dictionary_model_ind.keys():
+
+            if 'tasks' in self.question_dictionary_model_ind[question_key].keys():
+
+                # Get list of tasks for the question
+                question_tasks = self.question_dictionary_model_ind[question_key]['tasks']
+
+                # Check whether any of model_dep_tasks are in question_tasks
+                if any(elem in question_tasks for elem in all_tasks):
+
+                    # Ask the question
+                    self.ask_a_question(self.question_dictionary_model_ind, question_key)
+
+        # 9.2 Iterate over the model_dep dictionary and ask task questions
+        # ----------------------------------------------------------------
+        for model in self.experiment_dict['model_components']:
+
+            # Iterate over the model_dep dictionary and ask questions
+            # -------------------------------------------------------
+            for question_key in self.question_dictionary_model_dep[model].keys():
```

Comment:
Dict iterator already iterates over keys, so `.keys()` here is redundant.

```suggestion
            for question_key in self.question_dictionary_model_dep[model]:
```

Reply:
Hmm, I've been doing this all around the workflow, we should dedicate a `.keys[]` removal PR :)